capsules/extra/src/sha.rs

```diff
@@ -568,26 +539,9 @@ impl<
             .enter(processid, |app, kernel_data| {
                 match command_num {
                     // set_algorithm
-                    0 => {
-                        match data1 {
-                            // SHA256
-                            0 => {
-                                app.sha_operation = Some(ShaOperation::Sha256);
-                                CommandReturn::success()
-                            }
-                            // SHA384
-                            1 => {
-                                app.sha_operation = Some(ShaOperation::Sha384);
-                                CommandReturn::success()
-                            }
-                            // SHA512
-                            2 => {
-                                app.sha_operation = Some(ShaOperation::Sha512);
-                                CommandReturn::success()
-                            }
-                            _ => CommandReturn::failure(ErrorCode::NOSUPPORT),
-                        }
-                    }
+                    0 => match data1 {
+                        _ => CommandReturn::failure(ErrorCode::NOSUPPORT),
+                    },
 
                     // run
```

Comment:
If there is both run and finish, what does finish do?

Reply:
This is based on the Rust hash API: https://doc.rust-lang.org/std/hash/index.html